lib/razor/data/hook.rb

```diff
@@ -318,37 +318,38 @@ def exec_script(script, args)
       # Run the file from the hook's directory so that relative paths work.
       hook_dir = File.expand_path('..', script)
       stdin, stdout, stderr, wait_thr = Bundler.with_clean_env do
-        begin
-          old = Dir.pwd
-          extra_path = Razor.config['hook_execution_path']
-          ENV['PATH'] = "#{extra_path}:#{ENV['PATH']}" if extra_path
-          Dir.chdir(hook_dir)
+        extra_path = Razor.config['hook_execution_path']
+        ENV['PATH'] = "#{extra_path}:#{ENV['PATH']}" if extra_path
+        Dir.chdir(hook_dir) do
```

Comment:
Looking at JRuby's source (and chdir's specs), if chdir is passed a block, it will ensure that the old directory is restored after the code executes even if an exception is thrown. Here is the relevant part of the JRuby source:

https://github.com/jruby/jruby/blob/master/core/src/main/java/org/jruby/RubyDir.java#L310-L319

Also the docs on chdir: http://ruby-doc.org/core-2.2.3/Dir.html#method-c-chdir